src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -60,16 +90,71 @@ internal ConfigurationClientManager(IList<ConfigurationClientWrapper> clients)
         public IEnumerable<ConfigurationClient> GetAvailableClients(DateTimeOffset time)
         {
             return _clients.Where(client => client.BackoffEndTime <= time).Select(c => c.Client).ToList();
-        }
-
+        }
+
+        public async Task<IEnumerable<ConfigurationClient>> GetAutoFailoverClients(CancellationToken cancellationToken)
+        {
+            var isUsingConnectionString = _connectionStrings != null && _connectionStrings.Any();
+
+            Uri endpoint = null;
+            var secret = string.Empty;
+            var id = string.Empty;
+
+            if (isUsingConnectionString)
+            {
+                var connectionString = _connectionStrings.First();
+
+                endpoint = new Uri(ConnectionStringUtils.Parse(connectionString, ConnectionStringUtils.EndpointSection));
+                secret = ConnectionStringUtils.Parse(connectionString, ConnectionStringUtils.SecretSection);
+                id = ConnectionStringUtils.Parse(connectionString, ConnectionStringUtils.IdSection);
+            }
+            else
+            {
+                endpoint = _endpoints.First();
+            }
+
+            IEnumerable<SrvRecord> results = await _srvLookupClient.Value.QueryAsync(endpoint.DnsSafeHost, cancellationToken).ConfigureAwait(false);
+
+            // shuffle the results to ensure hosts can be picked randomly.
+            IEnumerable<string> srvTargetHosts = results.Select(r => $"{r.Target.Value.Trim('.')}").Shuffle().ToList();

```

Comment:
Do you mind shuffling on the non-transformed enumerable.

Reply:
What's that trim doing, just removing a preceding '.' ? If so, may be clearer to be explicit about that.

Reply:
@RichardChen820 

Why was this resolved? Did other comments get resolved without being considered?